### PR TITLE
parse w3c headers by default

### LINKED ISSUE #188: Enable Beeline to simultaneously accept w3c and honeycomb propagation headers
**Is your feature request related to a problem? Please describe.**

Currently, it's not possible to incrementally move a microservices architecture from the honeycomb propagation headers format to the w3c propagation headers format. This makes it difficult if not impossible for beeline customers with a large enough deployment to move towards interoperating with OpenTelemetry.

**Describe the solution you'd like**

Allow the beeline to simultaneously accept both w3c and honeycomb propagation headers, whichever is found on the incoming request. The outgoing propagation header format can still be static and set by configuration.

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by vreynolds**
## Which problem is this PR solving?

- closes #188 

## Short description of the changes

- preference: trace parser hook > honeycomb header > w3c header

### CODE REVIEW

File: beeline/trace.py [hunk 1]
```diff
@@ -49,8 +50,8 @@ def __init__(self, client):
 
         self.presend_hook = None
         self.sampler_hook = None
-        self.http_trace_parser_hook = beeline.propagation.honeycomb.http_trace_parser_hook
-        self.http_trace_propagation_hook = beeline.propagation.honeycomb.http_trace_propagation_hook
+        self.http_trace_parser_hook = beeline.propagation.default.http_trace_parser_hook
+        self.http_trace_propagation_hook = beeline.propagation.default.http_trace_propagation_hook
```
Comment (robbkidd) [inline beeline/trace.py:54]: Ooo. I didn't add an outgoing hook to Ruby's new default propagator. That's a good idea. 👍🏻

**Comment by robbkidd** [comment]
Is there still seeing it in EGS to do?

**Comment by vreynolds** [comment]
> Is there still seeing it in EGS to do?

I checked it in the greeting service ✅

**Review by robbkidd — APPROVED** [review]
Supports two headers at once!

Supports two headers at once!

**Pull request merged**